src/main/java/com/google/devtools/build/lib/packages/RuleTransitionData.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.packages;
+
+import com.google.auto.value.AutoValue;
+import com.google.devtools.build.lib.analysis.config.transitions.TransitionFactory;
+
+/**
+ * Helper class which contains data used by a {@link TransitionFactory} to create a transition for
+ * rules and attributes.
+ */
+// This class is in lib.packages in order to access AttributeMap, which is not available to
+// the lib.analysis.config.transitions package.
```

Comment:
Can we just add this as dependency in the BUILD file? We have references to `AttributeMap` throughout `lib.analysis`, so I don't see the harm in supporting it in `lib.analysis.config.transitions` too.

Reply:
No, because analysis.config.transitions is too low-level, and it leads to a circular dependency. I'd have to break AttributeMap out of build-base, and a dozen other classes, and it's too much work.

The entire Bazel package structure is a huge mess.

Reply:
Okay, no worries.